types/webpack/index.d.ts

```diff
@@ -1956,6 +1975,38 @@ declare namespace webpack {
         }
     }
 
+    namespace Template {
+        function getFunctionContent(fn: (...args: any[]) => any): string;
+
+        function toIdentifier(str: string): string;
+
+        function toComment(str: string): string;
+
+        function toNormalComment(str: string): string;
+
+        function toPath(str: string): string;
+
+        function numberToIdentifer(n: number): string;
+
+        function indent(s: string | string[]): string;
+
+        function prefix(s: string | string[], prefix: string): string;
+
+        function asString(str: string | string[]): string;
+
+        function getModulesArrayBounds(modules: {
+            id: string | number;
+        }): [number, number] | false;
+
+        function renderChunkModules(
+            chunk: compilation.Chunk,
+            filterFn: (module: Module, num: number) => boolean,
```

Comment:
`compilation.Module`? Otherwise it picks up the type from the global namespace (which `node` provides).